internal/http_api/api_request.go

```diff
@@ -19,12 +19,10 @@ type deadlinedConn struct {
 }
 
 func (c *deadlinedConn) Read(b []byte) (n int, err error) {
-	c.Conn.SetReadDeadline(time.Now().Add(c.Timeout))
 	return c.Conn.Read(b)
 }
 
 func (c *deadlinedConn) Write(b []byte) (n int, err error) {
-	c.Conn.SetWriteDeadline(time.Now().Add(c.Timeout))
```

Comment:
Right, sorry about the back and forth. What I realized is that, when `DeadlineTransport` is used for proxying (`nsqadmin`) there is no other request timeout being applied (unlike when it's used with an HTTP client).

We could set the `ResponseHeaderTimeout` only in the case where it's used for proxying?